packages/rum-core/src/domain/tracing/tracer.ts

```diff
@@ -122,6 +173,58 @@ function makeTracingHeaders(traceId: TraceIdentifier, spanId: TraceIdentifier, t
   }
 }
 
+/**
+ * https://www.w3.org/TR/trace-context/
+ */
+function makeTracingHeadersForW3C(
+  traceId: TraceIdentifier,
+  spanId: TraceIdentifier,
+  traceSampled: boolean
+): TracingHeaders {
+  return {
+    traceparent: `00-0000000000000000${traceId.toPaddedHexadecimalString()}-${spanId.toPaddedHexadecimalString()}-0${
+      traceSampled ? '1' : '0'
+    }`,
+  }
+}
+
+/**
+ * https://github.com/openzipkin/b3-propagation
+ */
+function makeTracingHeadersForB3(
+  traceId: TraceIdentifier,
+  spanId: TraceIdentifier,
+  traceSampled: boolean
+): TracingHeaders {
+  return {
+    b3: `${traceId.toPaddedHexadecimalString()}-${spanId.toPaddedHexadecimalString()}-${traceSampled ? '1' : '0'}`,
+  }
+}
+function makeTracingHeadersForB3M(
+  traceId: TraceIdentifier,
+  spanId: TraceIdentifier,
+  traceSampled: boolean
+): TracingHeaders {
+  return {
+    'X-B3-TraceId': traceId.toPaddedHexadecimalString(),
+    'X-B3-SpanId': spanId.toPaddedHexadecimalString(),
+    'X-B3-Sampled': traceSampled ? '1' : '0',
+  }
+}
+
+const makeTracingHeadersFor: {
+  [key in TracingHeadersType]: (
+    traceId: TraceIdentifier,
+    spanId: TraceIdentifier,
+    traceSampled: boolean
+  ) => TracingHeaders
+} = {
+  dd: makeTracingHeadersForDD,
+  w3c: makeTracingHeadersForW3C,
+  b3: makeTracingHeadersForB3,
+  b3m: makeTracingHeadersForB3M,
+} as const
+
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏keep it simple. TS will yell if any trace header is missing when using this object anywary

```suggestion
const makeTracingHeadersFor = {
  dd: makeTracingHeadersForDD,
  w3c: makeTracingHeadersForW3C,
  b3: makeTracingHeadersForB3,
  b3m: makeTracingHeadersForB3M,
}
```